src/scripts.js

```diff
@@ -2,107 +2,209 @@ import './css/styles.css';
 const guestPicker = document.querySelector('.guests-picker')
+const datePicker = document.querySelector('.date-picker')
+const button = document.querySelector('.submit-button')
+const messageBanner = document.querySelector('h3')
+const doubleClickMsg = document.querySelector('h4')
+const form = document.querySelector('.selections')
 
+//Event Listeners
+destinationsPicker.addEventListener('input', checkInput)
+datePicker.addEventListener('input', () => {checkDate(), checkInput()})
+durationPicker.addEventListener('input', checkInput)
+guestPicker.addEventListener('input', checkInput)
 
-const displayedUsersID = Math.floor(Math.random() * 50)
+button.addEventListener('click', checkNumberOfClick)
+window.addEventListener('load', buildFormHelper)
 
-window.addEventListener('load', buildForm)
 
-function buildForm() {
-  preventPickingPastDates()
-  buildDropDown()
-}
+const displayedUsersID = Math.floor(Math.random() * 50)
+let allDestinations;
+let allTrips;
+let allTravelers;
+let newPendingTrip;
+let today;
+
 
 Promise.all([travelersData, destinationsData, tripsData])
   .then((data) => {
     travelerHelper(data[0].travelers);
-    tripsHelper(data[2].trips, data[1].destinations);
+    destinationsHelper(data[1].destinations);
+    tripsHelper(data[2].trips);
   })
-  .catch((error) => console.log(error));
+  .catch((error) => {console.log(error), displayErrorMessage()});
+
+
+//HELPERS
 
+function buildFormHelper() {
+  preventPickingPastDates()
+  buildDropDown()
+  displayDestinationOptions(allDestinations.destinationsRepo)
+}
 
 function travelerHelper(data) {
-  const travelRepo = new Travelers(data)
-  travelerName.innerText = `Welcome, ${travelRepo.findTraveler(displayedUsersID)}`
+  allTravelers = new Travelers(data)
+  travelerName.innerText = `Welcome, ${allTravelers.findTraveler(displayedUsersID)}`
 }
 
-function tripsHelper(tripsData, destinationsData){
-  const tripsRepo = new Trips(tripsData)
-  const tripsPerTraveler = tripsRepo.getAllTrips(displayedUsersID)
-  
-  const allDestinations = new Destinations(destinationsData)
-  const totalCostThisYear = tripsRepo.totalCostPerYear(tripsPerTraveler, allDestinations)
+function destinationsHelper(destinationsData) {
+  allDestinations = new Destinations(destinationsData)
+  displayDestinationOptions(allDestinations.destinationsRepo)
+}
 
-  displayAllTrips(tripsPerTraveler, allDestinations, tripsRepo)
+function tripsHelper(tripsData){
+  allTrips = new Trips(tripsData)
+  console.log('alltrips', allTrips)
+  const tripsPerTraveler = allTrips.getAllTrips(displayedUsersID)
+  const totalCostThisYear = allTrips.totalCostPerYear(tripsPerTraveler, allDestinations)
+ 
+  displayAllTrips(tripsPerTraveler)
   displayTotalCostThisYear(totalCostThisYear)
-  displayDestinationOptions(allDestinations.destinationsRepo)
 }
 
-function displayAllTrips(tripsPerTraveler, allDestinations, tripsRepo) {
 
+//DISPLAY ALL TRIPS
+
+function displayAllTrips(tripsPerTraveler) {
```

Comment:
This function is great but be sure to refactor out some of the logic to make for better SRP. I am sure that is already in the plans but extracting the logic to separate files like (util or domManipulation) will score you a lot of points with your project manager